packages/discord.js/typings/index.d.ts

```diff
@@ -2986,6 +2986,161 @@ export const version: string;
 
 //#endregion
 
+//#region Errors
+export enum DiscordjsErrorCodes {
+  ClientInvalidOption,
+  ClientInvalidProvidedShards,
+  ClientMissingIntents,
+  ClientNotReady,
+
+  TokenInvalid,
+  TokenMissing,
+  ApplicationCommandPermissionsTokenMissing,
+
+  WSCloseRequested,
+  WSConnectionExists,
+  WSNotOpen,
+  ManagerDestroyed,
+
+  BitFieldInvalid,
+
+  ShardingInvalid,
+  ShardingRequired,
+  InvalidIntents,
+  DisallowedIntents,
+  ShardingNoShards,
+  ShardingInProcess,
+  ShardingInvalidEvalBroadcast,
+  ShardingShardNotFound,
+  ShardingAlreadySpawned,
+  ShardingProcessExists,
+  ShardingWorkerExists,
+  ShardingReadyTimeout,
+  ShardingReadyDisconnected,
+  ShardingReadyDied,
+  ShardingNoChildExists,
+  ShardingShardMiscalculation,
+
+  ColorRange,
+  ColorConvert,
+
+  InviteOptionsMissingChannel,
+
+  ButtonLabel,
+  ButtonURL,
+  ButtonCustomId,
+
+  SelectMenuCustomId,
+  SelectMenuPlaceholder,
+  SelectOptionLabel,
+  SelectOptionValue,
+  SelectOptionDescription,
+
+  InteractionCollectorError,
+
+  FileNotFound,
+
+  UserBannerNotFetched,
+  UserNoDMChannel,
+
+  VoiceNotStageChannel,
+
+  VoiceStateNotOwn,
+  VoiceStateInvalidType,
+
+  ReqResourceType,
+
+  ImageFormat,
+  ImageSize,
+
+  MessageBulkDeleteType,
+  MessageNonceType,
+  MessageContentType,
+
+  SplitMaxLen,
+
+  BanResolveId,
+  FetchBanResolveId,
+
+  PruneDaysType,
+
+  GuildChannelResolve,
+  GuildVoiceChannelResolve,
+  GuildChannelOrphan,
+  GuildChannelUnowned,
+  GuildOwned,
+  GuildMembersTimeout,
+  GuildUncachedMe,
+  ChannelNotCached,
+  StageChannelResolve,
+  GuildScheduledEventResolve,
+  FetchOwnerId,
+
+  InvalidType,
+  InvalidElement,
+
+  MessageThreadParent,
+  MessageExistingThread,
+  ThreadInvitableType,
+
+  WebhookMessage,
+  WebhookTokenUnavailable,
+  WebhookURLInvalid,
+  WebhookApplication,
+  MessageReferenceMissing,
+
+  EmojiType,
+  EmojiManaged,
+  MissingManageEmojisAndStickersPermission,
+  NotGuildSticker,
+
+  ReactionResolveUser,
+
+  VanityURL,
+
+  InviteResolveCode,
+
+  InviteNotFound,
+
+  DeleteGroupDMChannel,
+  FetchGroupDMChannel,
+
+  MemberFetchNonceLength,
+
+  GlobalCommandPermissions,
+  GuildUncachedEntityResolve,
+
+  InteractionAlreadyReplied,
+  InteractionNotReplied,
+  InteractionEphemeralReplied,
+
+  CommandInteractionOptionNotFound,
+  CommandInteractionOptionType,
+  CommandInteractionOptionNoSubcommand,
+  CommandInteractionOptionNoSubcommandGroup,
+  AutocompleteInteractionOptionNoFocusedOption,
+
+  ModalSubmitInteractionFieldNotFound,
+  ModalSubmitInteractionFieldType,
+
+  InvalidMissingScopes,
+
+  NotImplemented,
+
+  SweepFilterReturn,
+}
+
+export class DiscordjsError extends Error {
+  public override readonly name: string;
+  public readonly code: DiscordjsErrorCodes;
```

Comment:
But I do wonder if we can somehow get just the string keys of the enum

```suggestion
  public get code(): DiscordjsErrorCodes;
```

Reply:
We can, `keyof typeof DiscordjsErrorCodes`, but we don't really want that I believe, since that way TS users could hardcode strings to check against the property, rather than being forced to compare against `DiscordjsErrorCodes.SomeCode`.

Reply:
But this getter literally returns a string, not a number, so TS users could never actually validate that

Reply:
Oh, right. I guess `keyof typeof` it is.